schemachange/schemachange.c

```diff
@@ -671,27 +671,57 @@ int live_sc_post_delete(struct ireq *iq, void *trans, unsigned long long genid,
     return rc;
 }
 
+/* If the schema change is to 1) remove ODH, 2) change the compression
+   algorithm, 3) or alter a field from or to vutf8, unpack the blobs. */
 static int unodhfy_if_necessary(struct ireq *iq, blob_buffer_t *blobs,
                                 int maxblobs)
 {
-    int i, rc, oldodh, newodh, reccompr, oldcompr, newcompr;
-
-    for (i = 0, rc = 0; i != maxblobs; ++i) {
-        bdb_get_compr_flags(iq->usedb->sc_from->handle, &oldodh, &reccompr,
-                            &oldcompr);
-        bdb_get_compr_flags(iq->usedb->sc_to->handle, &newodh, &reccompr,
-                            &newcompr);
-        (void)reccompr;
-
-        /* If we're removing the ODH, or changing the compression algorithm,
-           unpack the blobs. */
-        if ((oldodh && !newodh) || oldcompr != newcompr) {
+    int i, rc, oldodh, newodh, reccompr, oldcompr, newcompr, fromidx, blobidx;
+    struct schema *from, *to;
+
+    bdb_get_compr_flags(iq->usedb->sc_from->handle, &oldodh, &reccompr,
+                        &oldcompr);
+    bdb_get_compr_flags(iq->usedb->sc_to->handle, &newodh, &reccompr,
+                        &newcompr);
+    (void)reccompr;
```

Comment:
Not your change, but, what's this for?

Reply:
The old logic basically says, "for each blob in the blob list, if the schema change is to change the compression algorithm, or to remove ODH, unpack the blob." Note that the same check is done over and over again inside the for-loop. I've placed the check before the for-loop. Hence the new logic now says," if the schema change is to change the compression algorithm or to remove ODH, unpack each blob in the blob list."